File: Python/349_Intersection_of_Two_Arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        d = {}
        l = []
        for i in nums1:
            if i not in d:
                d[i] = 1
        for j in nums2:
            if j in d:
                l.append(j)
                del d[j]
        return l

File: C++/349_Intersection_of_Two_Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> a;
        unordered_set<int> b;
        vector<int> ans;
        for(int &x : nums1){
            if(a.count(x) == 0) a.insert(x);
        }
        for(int &y : nums2){
            if(a.count(y) == 1 && b.count(y) == 0){
                b.insert(y);
                ans.push_back(y);
            }
        }
        return ans;
    }
};